research/guarded_ai.py

```
@@ -275,6 +369,10 @@ def simulate_activity(yaml_file_path):
                     if key in metadata:
                         del metadata[key]
 
+            # Handle metadata_clear - clear all metadata if set to True
+            if "metadata_clear" in transition and transition["metadata_clear"] == True:
```

Ruff (0.12.2)

Avoid equality comparisons to `True`; use `transition["metadata_clear"]:` for truth checks

Replace with `transition["metadata_clear"]`

(E712)